### PR TITLE
Refactor and add options

### PR DESCRIPTION
**Pull request by ghost**
So I did some more work, and I think this is a lot better. What do you think?

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,88 +1,117 @@
...
-    duo = duo.installTo(opts.installTo);
+  if (args && args.options) {
+    options = args.options;
   }
 
-  // TODO nests assets incorrectly, doesn't update paths correctly
-  if (!!opts.buildTo) {
-    duo = duo.buildTo(opts.buildTo);
+  /**
+   * TODO What's going on here? Needs documentation!
```
Comment (ghost) [inline index.js]: Could you clarify what's going on in this function? What does it do exactly?

**Comment by sergeyt** [comment]
@ismay I think your changes make code more clean now.

**Comment by ghost** [comment]
Cool! Also, what exactly does [this function](https://github.com/superwolff/gulp-duo/blob/ismay/options/index.js#L73) do?

File: index.js (hunk 1 above)
Comment (sergeyt) [inline index.js]: The code below loads duo plugins, but I think it is better that developer call `require` explicitly, I am sure I've done this by laziness. So I suggest to remove this code at all. But it is better to check that `plugins` is array of functions.

**Comment by sergeyt** [comment]
I've added a note about unclear function to https://github.com/sergeyt/gulp-duo/pull/4/files

File: index.js (hunk 1 above)
Comment (sergeyt) [inline index.js]: So the usage of duo with plugins could look like:

``` js
var babel = require('duo-babel');
var jade = require('duo-jade');
var myth = require('duo-myth');
var duo = require('gulp-duo')({
  plugins: [
    babel(),
    jade(),
    // adding custom plugin configuration
    myth({
      features: {
        import: false
      }
    })
 ]
})
```

example is taken from https://github.com/duojs/duo/blob/master/docs/plugins.md

**Comment by ghost** [comment]
1. I agree, I think the user should do the requiring of the plugins. So I removed the function that did the automatic requiring.
2. Also I'm checking if `options.plugins` is an array, and rewrote the `foreach` to a `for` loop.
3. Why should we check if the `plugins` array contains functions? We can also just leave it to the user to make sure it contains functions. What do you think?

**Comment by sergeyt** [comment]
re 3: agree no need to check here since duo can do that raising error

**Comment by ghost** [comment]
I updated the readme for the new functionality, and updated the examples to regular javascript (I think it will be easier for people to understand). I think this PR is ready. What do you think, could you check it for mistakes?

**Comment by sergeyt** [comment]
@ismay the PR looks good to me. I have minor point to remove use of `!!` in setOptions and replace `typeof options.concurrency !== 'undefined'` to just `options.concurrency`.

**Comment by ghost** [comment]
I simplified it some more, what do you think?

**Comment by sergeyt** [comment]
@ismay great cleanup! I'll merge now.

**Comment by ghost** [comment]
Cool, thanks!

**Pull request merged**